dotnet/src/SemanticKernel.Core/Memory/TextRag/TextRagDocument.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel.Memory.TextRag;
+
+/// <summary>
+/// Represents a document that can be used for Retrieval Augmented Generation (RAG).
+/// </summary>
+[Experimental("SKEXP0130")]
+public sealed class TextRagDocument
```

Comment:
I would avoid `Rag` as part of a model. Do we need to have this in core? Could just be an example/sample we can point to?

```suggestion
public sealed class TextRetrievalDocument
```

Reply:
Thanks, renamed to TextSearchDocument as discussed in meeting.